python/paddle/tensor/manipulation.py

```diff
@@ -2492,7 +2492,55 @@ def _var_to_list(var):
     out = x.matmul(y).reshape(shape_out)
     return out
 
+def take_along_axis(arr, indices, axis):
+    broadcast_shape_list = list(arr.shape)
+    broadcast_shape_list[axis] = 1
+    broadcast_shape = tuple(broadcast_shape_list)
+    if in_dygraph_mode():
+        indices = paddle.broadcast_to(indices, broadcast_shape)
+        return _C_ops.take_along_axis(arr, indices, 'Axis', axis)
+    check_variable_and_dtype(
+        arr, 'x',
+        ['float16', 'float32', 'float64', 'int32', 'int64', 'uint8'], 'take_along_axis')
+    check_variable_and_dtype(indices, 'index', ['int32', 'int64'], 'take_along_axis')
+    indices = paddle.broadcast_to(indices, broadcast_shape) # broadcast to shape of the input array first. 
```

Comment:
Oh I see it! broadcasting in not down in the take along axis operator, but a separate operator here.